pytorch_lightning/utilities/dataloader_fetcher.py

```diff
@@ -0,0 +1,134 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Generator, List, Tuple
+
+import torch
+
+import pytorch_lightning as pl
+
+
+def profiled_iterator(iterable, profiler):
+    iterator = iter(iterable)
+    while True:
+        try:
+            with profiler.profile("get_train_batch"):
+                yield next(iterator)
+        except StopIteration:
+            return
+
+
+class LightningFetcher:
```

Comment:
n00b question: certain classes are prefixed with `Lightning` while others aren't.
Components with name:
- LightningModule, LightningDataModule, LightningLoggerBase
- without: Trainer, Callbacks, Logger, Profiler

utilities like this don't contain logic specific to the rest of the framework, so I wonder if we could call this just Prefetcher?

it can also help users feel like they're not needing to learn lightning-specific things

Reply:
I would propose `DataFetcher`.